includes/admin/class-sensei-course-pre-publish-panel.php

```diff
@@ -77,6 +79,33 @@ public function maybe_publish_lessons( $course_id ) {
 			return;
 		}
 
+		$publishing_meta_key = '_sensei_course_publishing_started';
+
+		// Even if the course is already published, each subsequent updates also triggers this hook anyway
+		// which caused the bug https://github.com/Automattic/sensei/issues/7555.
+		// So we need to check if it's an actual publish call.
+		$is_main_publish_call = 'publish' !== $old_status;
+
+		if ( $is_main_publish_call ) {
+			// This is the first call made, it's not the structure saving call, so the added/updated lessons are not yet saved at this point.
+			// So we set the flag to publish lessons on the next call, which is made after the structure is saved.
+			update_post_meta( $course_id, $publishing_meta_key, true );
```

Comment:
Should we be returning here? Because otherwise it tries to publish the lessons which I guess doesn't do anything because they're not created yet? 😅

Reply:
Actually if you are trying to publish an already created draft course with existing draft lessons (or you've clicked on "Save Draft" or "Save to Edit Lesson" button already), they're already created before these calls. But in case you are publishing a brand new Course with Unsaved lesson, in that case they won't exist